vm/compiler/src/ledger/mod.rs

```diff
@@ -594,11 +591,38 @@ impl<N: Network, B: BlockStorage<N>, P: ProgramStorage<N>> Ledger<N, B, P> {
         Ok(())
     }
 
+    /// Adds a given address to the validator set.
+    pub fn add_validator(&mut self, address: Address<N>) -> Result<()> {
+        if self.validators().contains_key(&address) {
```

Comment:
since `insert` indicates whether the key had already existed and the value doesn't change, there's no need to use `contains_key`

Reply:
Logically, we wouldn't even need the `Result`s  here, but since the validator logic will be quite sensitive in subsequent testnet3 phases, I wanted to have dedicated error handling here.

Reply:
the `Result` is fine; what I'm suggesting is:
```
pub fn add_validator(&mut self, address: Address<N>) -> Result<()> {
    if self.validators.insert(address, ()).is_some() {
        bail!("'{address}' is already in the validator set.")
    } else {
        Ok(())
    }
}
```